### PR TITLE
Social | Following

### PR DESCRIPTION
**Pull request by bkoval**
@Wikia/social-backend @Wikia/social-frontend @Wikia/social-qa

### CODE REVIEW

File: front/main/app/templates/components/discussion-entities-list-layout.hbs [hunk 1]
```diff
@@ -12,6 +12,9 @@
 			smartBannerVisible=smartBannerVisible
 			updateCategories=(action this.attrs.updateCategories)
 			validatePostsOnForum=(action this.attrs.validatePostsOnForum)
+			isSortingDisabled=isSortingDisabled
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-entities-list-layout.hbs]: 🚓

File: front/main/app/templates/components/discussion-entities-list-layout.hbs [hunk 2]
```diff
@@ -53,11 +58,17 @@
...
+			{{#if canShowReportedPostsFilter}}
 				{{discussion-reported-filter
 					applyFilters=(action this.attrs.applyFilters)
+					topDecoration=reportedFilterTopDecoration
 				}}
 			{{/if}}
+			{{discussion-sort
+					sortBy=sortBy
+					setSortBy=(action 'setSortBy')
+					disabled=isSortingDisabled
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-entities-list-layout.hbs]: 🚓

File: front/main/app/templates/components/discussion-tooltip.hbs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{{#if isVisible}}
+	<div class="discussion-tooltip active-element-background-color">
+		{{text}}
+	</div>
+	<div class="arrow arrow-{{arrowDirection}} active-element-border-theme-color" style={{arrowStyle}}></div>
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-tooltip.hbs]: I think the rule of thumb is to use <span> elements for empty-content elements.

File: front/main/app/templates/components/discussion-header.hbs [hunk 4]
```diff
@@ -17,22 +17,23 @@
 			</div>
 		{{/if}}
 		{{#if hasFilters}}
-			<div class="sort {{if isFilterApplied 'applied'}}" id="{{discussionMobileFilterId}}">
+			<div class="sort{{if isFilterApplied ' applied'}}" id="{{discussionMobileFilterId}}">
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-header.hbs]: Why this change? Wasn't original version working fine? :)

File: front/main/app/templates/components/discussion-introducing-following.hbs [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<div class="introducing-following">
+	<header class="background-theme-color">
+		{{svg 'follow-small' viewBox='0 0 32 42' class='follow-icon'}}
+	</header>
+	<section>
+		<h6>{{i18n 'main.follow-introducing-header' ns='discussion'}}</h6>
+		{{i18n 'main.follow-introducing-text' ns='discussion'}}
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-introducing-following.hbs]: This will have no html node around it and the rule is to prevent having text nodes mixed with html elements. How about a paragraph tag here? Or at least a span :)

File: front/main/app/templates/components/discussion-no-followed-posts.hbs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{{svg 'follow-small' viewBox='0 0 32 42' class='follow-xl-icon'}}
+<div class="header">
+	{{i18n "main.no-followed-posts-header" ns="discussion"}}
+</div>
+<div>
+	{{i18n "main.no-followed-posts-text" ns="discussion"}}
+</div>
+{{link-to (i18n 'main.find-posts-to-follow' ns='discussion') 'discussion.forum' class='find-posts-button active-element-theme-color active-element-border-theme-color'}}
+
+
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-no-followed-posts.hbs]: I think we should get rid of these emepty lines :)

File: front/main/app/templates/components/discussion-no-followed-posts.hbs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{{svg 'follow-small' viewBox='0 0 32 42' class='follow-xl-icon'}}
+<div class="header">
```
Comment (bkoval) [inline front/main/app/templates/components/discussion-no-followed-posts.hbs]: Header as a div element? :)

File: front/main/app/mixins/discussion-contribution-model.js [hunk 8]
```diff
@@ -168,6 +169,54 @@ export default Ember.Mixin.create({
...
+
+		return request(M.getFollowingServiceUrl(endpoint), {
+			data: JSON.stringify({
+				siteId: Mercury.wiki.id
+			}),
+			method
+		}).then((data) => {
+			track(isFollowed ? trackActions.UnfollowPost : trackActions.FollowPost);
+		}).catch(() => {
+			entity.set('isFollowed', isFollowed);
```
Comment (mleczey) [inline front/main/app/mixins/discussion-contribution-model.js]: In case when follow action will not succeed I want to revert to state before hitting follow button.

File: front/main/app/components/discussion-more-options.js [hunk 9]
```diff
@@ -43,6 +47,16 @@ export default Ember.Component.extend({
 	},
 
 	actions: {
+		/**
+		 * @param {Object} post
+		 *
+		 * @returns {void}
+		 */
+		delete(post) {
+			this.get('delete')(post);
```
Comment (mleczey) [inline front/main/app/components/discussion-more-options.js]: Added

File: front/main/app/mixins/discussion-forum-model-static.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import Ember from 'ember';
```
Comment (latata) [inline front/main/app/mixins/discussion-forum-model-static.js]: It contains static methods for model class
